src/layers/pip_cache.rs

```diff
@@ -0,0 +1,80 @@
+use crate::packaging_tool_versions::PackagingToolVersions;
+use crate::python_version::PythonVersion;
+use crate::PythonBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::data::buildpack::StackId;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::Buildpack;
+use libherokubuildpack::log::log_info;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+/// Layer containing Pip's cache of HTTP requests/downloads and built package wheels.
+pub(crate) struct PipCacheLayer<'a> {
```

Comment:
`Debug` all structs by default please.

Reply:
Not having it makes println and dbg based debugging harder, and (in my opinion) surprising. If I'm in your codebase and need it to debug something, and add it to one struct, then it has to cascade to all other structs that it holds that also don't implement debug. Then once I finally fix the bug I was chasing, I'm left with two options:

- Submit a messy PR with a bunch of Debug additions and my fix.
- Carefully remove all the Debug additions and hope that I don't have to add them back in after someone realizes there's an edge case that needs to be handled.

With Rust print/puts debugging is less needed, but it's not an absurd or exotic use case.

Reply:
I agree that is one case where having `Debug` on internal things can be useful.

However:
- I rarely use println debugging (instead unit test, or the debugger)
- It very easy to exclude print statements or things added when developing using partial Git stating (I use GitKraken, but there are lots of ways to do this)
- The structs in this buildpack aren't super complicated/nested, so the "I need to cascade Debug everywhere" issue doesn't apply here

If this ever comes up in reality as being a problem for this buildpack, then I'm happy to reconsider :-)